modules/channel-web/src/backend/db.ts

```diff
@@ -1,23 +1,102 @@
-import Bluebird from 'bluebird'
 import * as sdk from 'botpress/sdk'
+import { Raw } from 'knex'
 import _ from 'lodash'
 import moment from 'moment'
 import ms from 'ms'
 import uuid from 'uuid'
 
 import { Config } from '../config'
 
+import { DBMessage } from './typings'
+
 export default class WebchatDb {
-  knex: any
-  users: typeof sdk.users
+  private readonly MAX_RETRY_ATTEMPTS = 3
+  private knex: sdk.KnexExtended
+  private users: typeof sdk.users
+  private batchedMessages: DBMessage[] = []
+  private batchedConvos: Dic<Raw<any>> = {}
+  private messagePromise: Promise<void | number[]>
+  private convoPromise: Promise<void>
+  private batchSize: number
 
   constructor(private bp: typeof sdk) {
     this.users = bp.users
     this.knex = bp['database'] // TODO Fixme
+
+    this.batchSize = process.env.DATABASE === 'postgres' ? 2000 : 40
```

Comment:
I don't see where this environment variable is declared anywhere in the code. Is is really set when using postgres?

Reply:
It seems to be understood that this environment variable is needed for postgres ([tutorial 1](https://aabingunz.com/switching-from-sqlite-to-postgres-in-botpress/), [tutorial 2](https://abhisheksimon.wordpress.com/2019/12/15/switching-from-sqlite-to-postgres-in-botpress/)) although our [documentation](https://botpress.com/docs/tutorials/database) doesn't mention it so that's bizarre. When testing it does it fact seem to be unnecessary. An alternative would be to do `process.env.DATABASE_URL.startsWith('postgres')`

Reply:
Those are outdated, you'd be better off with knex.isLite